src/ucs/vfs/sock/vfs_sock.c

```diff
@@ -27,23 +32,25 @@ typedef struct {
 
 void ucs_vfs_sock_get_address(struct sockaddr_un *un_addr)
 {
-    struct passwd *pw;
-    uid_t euid;
-
     memset(un_addr, 0, sizeof(*un_addr));
     un_addr->sun_family = AF_UNIX;
+    ucs_fill_filename_template(ucs_global_opts.vfs_sock_path, un_addr->sun_path,
+                               sizeof(un_addr->sun_path));
+}
 
-    euid = geteuid();
-    pw   = getpwuid(euid);
-    if (pw != NULL) {
-        /* By name */
-        snprintf(un_addr->sun_path, sizeof(un_addr->sun_path) - 1,
-                 "/tmp/ucx-vfs-%s.sock", pw->pw_name);
-    } else {
-        /* By number */
-        snprintf(un_addr->sun_path, sizeof(un_addr->sun_path) - 1,
-                 "/tmp/ucx-vfs-%u.sock", euid);
+int ucs_vfs_sock_mkdir(const char *sock_path)
+{
+    char sock_path_dir[PATH_MAX];
+    int ret;
+
+    ucs_strncpy_safe(sock_path_dir, sock_path, sizeof(sock_path_dir));
+    ret = mkdir(dirname(sock_path_dir), S_IRWXU);
+    if ((ret < 0) && (errno != EEXIST)) {
+        ucs_diag("failed to create directory '%s': %m", sock_path_dir);
```

Comment:
maybe pass log_level as argument, so it will be ERROR when using from ucx_vfs tool?

Reply:
ok, sure